assets/js/components/notifications/module-settings-warning.js

```diff
@@ -1,16 +1,35 @@
 /**
- * WordPress dependencies
+ * Internal dependencies
  */
-import { withFilters } from '@wordpress/components';
-import { Component } from '@wordpress/element';
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import classnames from 'classnames';
+import ErrorIcon from '../../../svg/error.svg';
 
-/**
+const { useSelect } = Data;
+
+/*
  * A single module. Keeps track of its own active state and settings.
  */
-class ModuleSettingsWarning extends Component {
-	render() {
+export default function ModuleSettingsWarning( {
+	slug,
+	// context,
+} ) {
+	// @TODO: Resolver only runs once per set of args, so we are working around
+	// this to rerun after modules are loaded.
+	// Once #1769 is resolved, we can remove the call to getModules,
+	// and remove the !! modules cache busting param.
+	const modules = useSelect( ( select ) => select( CORE_MODULES )?.getModules() );
+	const requirementsStatus = useSelect( ( select ) => select( CORE_MODULES )?.getCheckRequirementsStatus( slug, !! modules ) );
```

Comment:
How about this for an alternate approach: https://github.com/google/site-kit-wp/pull/2335/commits/ae94aa2cb24dd5fe5c948d862c21557139ab2039#diff-51079bbb133e20fa794c0e1fd5fd30b551dda818e0cfe0ba4162502670cf133aR382

Reply:
@felixarntz @ivankruchkoff there is a bit of an odd situation here that we haven't run into yet where we essentially need to wait for the module to be registered which can currently happen at any time. For our internal modules, these are all registered on `domReady` and that seems to work just fine to run before all the datastore stuff runs that depends on it like this. This may be something we can improve in a separate issue, but no changes to this are needed here – we just need to wait for modules to be loaded.

While some resolver invalidation may be necessary (depending on what happens in the resolver), the important thing to add here is a `resolveSelect` call to `getModules` for both `canActivateModule` and `getCheckRequirementsStatus` resolvers (they currently share the same resolver) but that way we can essentially wait for `getModule(s)` to return actual data and all modules should be registered at this point. This currently requires that we would use the new `await` utility action with `__experimentalResolveSelect` but this should be fine since this API is now stable in newer versions of `@wordpress/data`.

Reply:
Added the call to `__experimentalResolveSelect` in https://github.com/google/site-kit-wp/pull/2335/commits/dd9f2e780e3d5e1df3c35f124af89cb83fdc8f47